### PR TITLE
[FIX] web_editor: consider the header when scrolling to an element

### PR DESCRIPTION
**Pull request by sobo-odoo**
In commit [1], the `dom.js` functions linked to the scroll were converted in order to not use jQuery anymore. They were then later completely removed from this file in commit [2] and then added in `snippets.editor.js`, as well as some other functions that would be needed in this file.

However, the way the conversion was made did not take into account the fact that a header is present above the website pages and that this header could disappear and reappear during the scroll. Some parts were also not converted correctly. This resulted in some website behaviors being broken.

This commit fixes this conversion by:
- recomputing the scroll destination at each animation frame (as it was done before) to consider the header position and adapt the scroll to it.
- getting the correct `scrollElement` by using the `getScrollElement` util (like before) instead of doing `scrollDocument.scrollingElement`. Indeed, the scrolling element needs to be found recursively, which is why the element was not the correct one.
- adding a missing `window.scrollY` when converting one of the jQuery `offset()` call (like done everywhere else).

Steps to reproduce:
- In website edit mode, drop enough snippets to have a scrollbar.
- Drop another snippet at the top or in the middle of the page.
=> The screen will scroll to the snippet by taking into account a 50px extra offset, which is normal. However, this offset is added from the top of the page, instead of starting from the bottom of the header as it was done before.

[1]: https://github.com/odoo/odoo/commit/8c5a7dc788488baccf42f1a99d1e9107c03083ef
[2]: https://github.com/odoo/odoo/commit/77797a51bbbdc8d62ba44a984d546823f23f935a

task-4128905

### CODE REVIEW

File: addons/web/static/src/core/utils/website_scrolling.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
+// Scrolling util functions needed by the Website module and sub-modules. These
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/website_scrolling.js:1]: Putting this in core makes it available in the backend as well. According to its name, it looks like it's only necessary for website => move it to the web/static/src/public folder? Or directly in the website/portal addons if that works (would be better actually...).

Side-note: it's a bit disappointing to have several scrolling helpers...

Comment (qsm-odoo) [reply addons/web/static/src/core/utils/website_scrolling.js:1]: > Side-note: it's a bit disappointing to have several scrolling helpers...

It sure is, that's why we only had my implementation... before someone implemented new ones, changed and moved existing ones, breaking practically all our usecases... :shrug: Thanks for the list of bugs and the work @sobo-odoo -> Maybe you can create a task with that list so we can create tests about those issues later? As we will have to refactor those utils at some point :point_down: 

> Or directly in the website/portal addons if that works

@sobo-odoo Let's move that file in `web_editor` and rename the file `scrolling.js` :+1: The web_editor app is meant to be removed in the future I guess so we'll see how we can share utils without touching `web` at that point. Maybe after `#wrapwrap` removal and other stuff, we'll get closer to be able to use the official util... :shrug:

File: addons/website_blog/static/src/js/website_blog.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ publicWidget.registry.websiteBlog = publicWidget.Widget.extend({
         const recordCoverContainerEl = ev.currentTarget.querySelector(".o_record_cover_container");
         const classes = nexInfo.size.split(" ");
         recordCoverContainerEl.classList.add(...classes, nexInfo.textContent);
-        ev.currentTarget.querySelector(".o_wblog_toggle").classList.toggle("d-none");
+        ev.currentTarget.querySelectorAll(".o_wblog_toggle").forEach(el => el.classList.toggle("d-none"));
```
Comment (qsm-odoo) [inline addons/website_blog/static/src/js/website_blog.js:40]: FYI @robinlej :wink: 
Don't know how you caught that @sobo-odoo :open_mouth:

**Review by qsm-odoo — APPROVED** [review]
Thanks @sobo-odoo for the clarity and the details, I tested everything, looks good :+1: I only push-forced to not mention the word "website" in a web_editor file and to fix some commit message typos while I was at it :)

Let's go ahead with that, then I may be able to finish the wrapwrap scroll PR then we will finally be able to reconsider using official utils or extension of them next year.

@robodoo rebase-ff r+

**Pull request closed without merging**